markdown/events/2021/bytesize-5-dsl2-module-development.md

```diff
@@ -32,3 +32,257 @@ This week, Harshil Patel ([@drpatelh](http://github.com/drpatelh/)) will present
+
+[27:20](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1640)
+Similarly, these, these and these are all again initiated here in this initOptions map which just initiates decent values now. This works, it worked for the rna-seq pipeline, it may not be as comprehensive, it's slightly annoying where some tools require three arguments and so you have `options.args3` which is slightly ugly but it works until we have a better solution.
+
+[27:44](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1664)
+And this, here, is how we're saving files again, a couple of Groovy functions and that allows us to provide the files, as I showed you before, in this sort of format where you just have a Groovy map of which files you want to publish.
+
+Not everyone will want to publish all of the files. Some files they may want to publish in different ways.
+
+[28:08](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1688)
+Right, so that's just going back to the include statement and where these functions are included in the module. The name of the process is all lowercase, so this is the path to the process, it's samtools/sort. The process name must be the same as the module name, but all uppercase and, again, it should just be separated by a single underscore. This is quite important for standardization, as I mentioned to you before, we're getting the software name, we get the process name, so it'd be nice to sort of standardize this so we always get the main tool name that we can publish as a default.
+
+[28:44](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1724)
+And so, this is sort of a niggly thing, I mean there are obviously edge cases where you can have three layers of tools and so on, but this should work for 99.99% of cases.
+
+The tag, by default, uses this meta map of sample information and this is generally provided in the input section here (which i will come back to later) and so this just allows you, when Nextflow is running nicely in the terminal, to see which sample is running and it just gives it a tag as to as to what's running.
+
+[29:14](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1754)
+This may not be always possible because in some instances, where you're indexing genomes for example, you don't need any sample information so you just tag it with whatever is appropriate. Appropriately resource labels we have in the conf/base.config of nf-core pipelines and we just have some simple process labels - process low,  medium, high, maybe retry or something like that. This just allows us again to just reuse these labels across nf-core pipelines and even in your own Nextflow pipelines if you so wish to use them.
```

Comment:
```suggestion
This may not be always possible because in some instances, where you're indexing genomes for example, you don't need any sample information so you just tag it with whatever is appropriate. Appropriately resource labels we have in the conf/base.config of nf-core pipelines and we just have some simple process labels - process low, medium, high, maybe retry or something like that. This just allows us again to just reuse these labels across nf-core pipelines and even in your own Nextflow pipelines if you so wish to use them.
```